builtin/providers/scaleway/resource_scaleway_ip.go

```diff
@@ -0,0 +1,66 @@
+package scaleway
+
+import "github.com/hashicorp/terraform/helper/schema"
+
+func resourceScalewayIP() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayIPCreate,
+		Read:   resourceScalewayIPRead,
+		Update: resourceScalewayIPUpdate,
+		Delete: resourceScalewayIPDelete,
+		Schema: map[string]*schema.Schema{
+			"server": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceScalewayIPCreate(d *schema.ResourceData, m interface{}) error {
+	scaleway := m.(*Client).scaleway
+	resp, err := scaleway.NewIP()
+	if err != nil {
+		return err
+	}
+
+	d.SetId(resp.IP.ID)
+	return resourceScalewayIPUpdate(d, m)
+}
+
+func resourceScalewayIPRead(d *schema.ResourceData, m interface{}) error {
+	scaleway := m.(*Client).scaleway
+	resp, err := scaleway.GetIP(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.Set("ip", resp.IP.Address)
+	d.Set("server", resp.IP.Server.Identifier)
+	return nil
+}
+
+func resourceScalewayIPUpdate(d *schema.ResourceData, m interface{}) error {
+	scaleway := m.(*Client).scaleway
+	if d.HasChange("server") {
+		if err := scaleway.AttachIP(d.Id(), d.Get("server").(string)); err != nil {
```

Comment:
It should detach the server. I'll add an explicit test case to make sure this behaves as expected. Sadly the scaleway cli does not expose a top level function like `detachIP`